resources/views/vendor/pagination/custom.blade.php

```diff
@@ -0,0 +1,110 @@
+                    </li>
+                @else
+                    <li class="page-item">
+                        <a class="page-link" href="{{ $paginator->previousPageUrl() }}" rel="prev">@lang('pagination.previous')</a>
+                    </li>
+                @endif
+
+                {{-- Next Page Link --}}
+                @if ($paginator->hasMorePages())
+                    <li class="page-item">
+                        <a class="page-link" 
+                        href="{{ $paginator->nextPageUrl() }}" 
+                        rel="next">Next</a>
+                    </li>
+                @else
+                    <li class="page-item disabled" aria-disabled="true">
+                        <span class="page-link">@lang('pagination.next')</span>
+                    </li>
+                @endif
+            </ul>
+        </div>
+
+        <div class="d-none flex-sm-fill d-sm-flex align-items-sm-center justify-content-sm-between">
+            {{-- show in to --}}
+            <div>
+                <ul class="pagination">
+
+                    {{-- Previous Page Link --}}
+                    @if ($paginator->onFirstPage())
+                        <li class="page-item disabled" aria-disabled="true" aria-label="@lang('pagination.previousText')">
+                            <span class="page-link" href="{{ $paginator->url(1) }}">@lang('pagination.previousText')</span>
+                        </li>
+                        <li class="page-item disabled" aria-disabled="true" aria-label="@lang('pagination.previous')">
+                            <span class="page-link" aria-hidden="true">@lang('pagination.previous')</span>
+                        </li>
+                    @else
+                        <li class="page-item">
+                            <a class="page-link" href="{{$paginator->url(1)}}" >@lang('pagination.previousText')</a>
+                        </li>
+                        <li class="page-item">
+                            <a class="page-link" href="{{ $paginator->previousPageUrl() }}" rel="prev" aria-label="@lang('pagination.previous')">@lang('pagination.previous')</a>
+                        </li>
+                    @endif
+                    @php
+                    $startPage = 1;
+                    $endPage = 5;
+
+                    if($paginator->currentPage() > 4) {
+                        if($paginator->currentPage() > $paginator->lastPage() - 4) {
+                            $startPage = $paginator->lastPage() - 4;
+                        } else {
+                            $startPage = $paginator->currentPage() - 2;
+                        }
+                    } else {
+                        $startPage = 1;
+                    }
+
+                    $endPage = $startPage + 4;  
+                    @endphp
+                    {{-- Pagination Elements --}}
+                    @foreach ($elements as $element)
```

Comment:
Tại sao không xử lý dữ liệu chuẩn ở BE trc mà phải ra FE xử lý if/else lồng nhau nhiều ntn ?